### ISSUE TITLE
Group Column Disappears When Hiding Any Other Column

### ISSUE THREAD

**Issue by ambroselittle** [ceolter/ag-grid#1590]
**I'm submitting a ...**  (check one with "x")
```
[ x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

I configured a group column like so:
```
    groupSuppressAutoColumn: false,
    enableSorting: true,
    enableFilter: false,
    groupColumnDef: {
        cellRenderer: 'group', // ag-grid magic value to tell it to render this as the group col
        headerName: 'Receivables',
        pinned: 'left',
        minWidth: 150,
        suppressMovable: true,
        suppressResize: true,
        suppressSizeToFit: true
    },
```
Then I use the menu to go to the column chooser and deselect any other column I have. At that point, it hides the group column as well and never comes back. 

Vid: https://www.dropbox.com/s/trzt9o7al03qo6t/group-col-disappear-bug.mov?dl=0 

**Expected behavior**

Should not hide that column or impact it in any way.

**Minimal reproduction of the problem with instructions**

See above for group col def and the actual behavior in a vid demo.

**What is the motivation / use case for changing the behavior?**

Users may want to hide some of the columns to more easily compare values across select columns, i.e., standard reason to hide columns. :) 

**Please tell us about your environment:**

VS Code, mixed package manager, ASP.NET on Win 10 (server). It is running in React, too.

* **ag-Grid version:** X.X.X

v7.2.3

* **Browser:**

This is latest Chrome

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
ES6 transpiled.

**Comment by seanlandsman** [comment]
Hi

Thanks for reporting this - I've raised issue AG-414 our side and we'll update this thread as soon as we have any updates

thanks

**Comment by ambroselittle** [comment]
Yes, we're enterprise. I'll see when I can get back to this with a repro I can share.
